packages/syft/src/syft/custom_worker/builder.py

```diff
@@ -54,27 +95,42 @@ def build_image(self, config: CustomWorkerConfig) -> None:
             f"with args={build_args}"
         )
 
-        try:
-            # TODO: Push logs to mongo/seaweed?
-            with contextlib.closing(docker.from_env()) as client:
-                client.images.build(
-                    path=str(contextdir),
-                    dockerfile=dockerfile,
-                    pull=True,
-                    tag=f"{self.CUSTOM_IMAGE_PREFIX}-{type}:{imgtag}",
-                    timeout=self.BUILD_MAX_WAIT,
-                    buildargs=build_args,
+        return self._build_image(
+            tag=f"{self.CUSTOM_IMAGE_PREFIX}-{type}:{imgtag}",
+            path=str(contextdir),
+            dockerfile=dockerfile,
+            buildargs=build_args,
+        )
+
+    def _build_image(self, tag: str, **build_opts) -> Tuple[Image, Iterable[str]]:
```

Comment:
This should be just `Iterable` instead. From [`docker-py` API reference](https://docker-py.readthedocs.io/en/stable/images.html#docker.models.images.ImageCollection.build) "The second item is a generator of the build logs as JSON-decoded objects."